sklearn/cluster/_kmeans.py

```diff
@@ -564,29 +566,32 @@ def _kmeans_single_lloyd(X, sample_weight, n_clusters, max_iter=300,
         lloyd_iter = lloyd_iter_chunked_dense
         _inertia = _inertia_dense
 
-    for i in range(max_iter):
-        lloyd_iter(X, sample_weight, x_squared_norms, centers, centers_new,
-                   weight_in_clusters, labels, center_shift, n_threads)
-
-        if verbose:
-            inertia = _inertia(X, sample_weight, centers, labels)
-            print("Iteration {0}, inertia {1}" .format(i, inertia))
+    # Threadpoolctl context to limit the number of threads in second level of
+    # nested parallelism (i.e. BLAS) to avoid oversubsciption.
+    with threadpool_limits(limits=1, user_api="blas"):
```

Comment:
Ok so you confirm that we have no BLAS call that are protected by this CM, apart from the ones in `lloyd_iter`?

Reply:
Yes I just double checked